### DISCUSSION TITLE
Restoration failed and exiting of IPOPT

### DISCUSSION THREAD

**Discussion by LoukasKyriakidis** [coin-or/Ipopt#711] [General]
I am currently using Pyomo 6.4.2. with IPOPT 3.11.1 as a solver for a nonlinear constrained optimization problem in Python 3.8.12. in Windows 10. The optimization problem has a dimension of 408. When I run the optimization, I sometimes get the following error:

**Restoration phase converged to a point with small primal infeasibility

EXIT: Restoration Failed!

ValueError: Cannot load a SolverResults object with bad status: error**

This error does not appear every time I optimize this particular problem. I sometimes get this error and sometimes it works without any problem.

I read on internet that this indicates that the restoration phase failed to find a feasible point that was acceptable to the filter line search for the original problem. This could happen if the problem is highly degenerate, does not satisfy the constraint qualification, or if the NLP code provides incorrect derivative information. Consequently, IPOPT stops the optimization and exits instead of continuing with a new starting point.

In addition, I realised that I don't get this error if I optimize problems of lower dimension.

What I would like from IPOPT is to continue with a new starting point instead of exiting as it happens in GAMS.

**Error Message**

Restoration phase converged to a point with small primal infeasibility

EXIT: Restoration Failed!

ValueError: Cannot load a SolverResults object with bad status: error

**Information on the system**

Pyomo version: 6.4.2
Python version: 3.8.12
Operating system: Windows 10
How Pyomo was installed (PyPI, conda, source): conda
Solver (if applicable): ipopt

**Comment by svigerske** [comment]
Ipopt has nothing built-in to pick a different starting point and start again. (Also in GAMS there is nothing like that.)
You would have to implement this in your code that calls Pyomo.

First, I would suggest updating Ipopt though: https://github.com/coin-or/Ipopt/releases
3.11.1 is 10 years old!

Also check whether it is possible to improve your model.
Sometimes, changing the linear solver in Ipopt could also help.

**Reply by LoukasKyriakidis** [reply]
Thank you very much for your reply. 

A linear solver could help, but I need a nonlinear solver, I want to keep the problem nonlinear.

The reason why I am using this version for ipopt is that I tried to install a new one, but I faced some problems with the installation. For this reason I kept the older one.

The problem is that even if I implement something in my code that calls Pyomo, I receive the same error message. Maybe it has to do wiith the version. I am going to try the new one.

**Reply by paul-scott** [reply]
By "linear solver" I suspect the parent is referring to the backend linear solver that ipopt uses (mumps, ma27, etc). At the end of the day the the barrier / interior-point method needs to solve a sequence of difficult linear systems whether or not your model is nonlinear.
